res/controllers/Pioneer-DDJ-200-scripts.js

```diff
@@ -1,406 +1,594 @@
-};
-
-DDJ200.rateLSB = function(channel, control, value, status, group) {
-    var vDeckNo = DDJ200.vDeckNo[script.deckFromGroup(group)];
-    var vgroup = "[Channel" + vDeckNo + "]";
-    // calculate rate value from its most and least significant bytes
-    var rateMSB = DDJ200.vDeck[vDeckNo]["rateMSB"];
-    var rate = 1 - (((rateMSB << 7) + value) / 0x1FFF);
-    engine.setValue(vgroup, "rate", rate);
-};
-
-DDJ200.volumeMSB = function(channel, control, value, status, group) {
-    // store most significant byte value of volume
-    var vDeckNo = DDJ200.vDeckNo[script.deckFromGroup(group)];
-    DDJ200.vDeck[vDeckNo]["volMSB"] = value;
-};
+            padInstancesBuffer.next();
+        };
+        setPads(padInstancesBuffer.current());
+    };
 
-DDJ200.volumeLSB = function(channel, control, value, status, group) {
-    var vDeckNo = DDJ200.vDeckNo[script.deckFromGroup(group)];
-    var vgroup = "[Channel" + vDeckNo + "]";
-    // calculate volume value from its most and least significant bytes
-    var volMSB = DDJ200.vDeck[vDeckNo]["volMSB"];
-    var vol = ((volMSB << 7) + value) / 0x3FFF;
-    //var vol = ((volMSB << 7) + value); // use for linear correction
-    //vol = script.absoluteNonLin(vol, 0, 0.25, 1, 0, 0x3FFF);
-    engine.setValue(vgroup, "volume", vol);
-};
+    const updateDeckHelper = function(padInstance, newGroup) {
+        padInstance.forEachComponent(function(component) {
+            if (padInstance === choosenPadInstance) {
+                component.disconnect();
+            }
+            if (component.group === undefined) {
+                component.group = newGroup;
+            } else {
+                const anyChannelRegEx = /\[Channel\d+([\]_])/;
+                component.group = component.group.replace(anyChannelRegEx, `${newGroup.slice(0, -1)}$1`);
+            }
+            if (padInstance === choosenPadInstance) {
+                component.connect();
+                component.trigger();
+            }
+        });
+    };
 
-DDJ200.eq = function(channel, control, value, status, group) {
-    var val = script.absoluteNonLin(value, 0, 1, 4);
-    var eq = (control === 0x0B) ? 2 : 1;
-    if (control === 0x07) {
-        eq = 3;
-    }
-    var deckNo = group.substring(24, 25);
-    // var deckNo = group.match("hannel.")[0].substring(6); // more general
-    // var deckNo = script.deckFromGroup(group); // working after fix
-    // https://github.com/mixxxdj/mixxx/pull/3178 only
-    var vDeckNo = DDJ200.vDeckNo[deckNo];
-    var vgroup = group.replace("Channel" + deckNo, "Channel" + vDeckNo);
-    engine.setValue(vgroup, "parameter" + eq, val);
-};
+    this.updateDeck = function(currentDeck) {
+        padInstancesBuffer.indexable[0].forEach(function(padInstance) {
+            updateDeckHelper(padInstance, currentDeck);
+        });
+        padInstancesBuffer.indexable[0].forEach(function(padInstance) {
+            updateDeckHelper(padInstance, currentDeck);
+        });
+    };
```

Comment:
this just updates the first padInstance, but you want to update the current instance, right?